Sources_v3/StreamChatUI/ChatChannel/ChatVC.swift

```diff
@@ -150,7 +150,14 @@ open class ChatVC<ExtraData: ExtraDataTypes>: ViewController,
     public func chatMessageListVC(_ vc: ChatMessageListVC<ExtraData>, didTapOnRepliesFor message: _ChatMessage<ExtraData>) {}
 
     public func chatMessageListVC(_ vc: ChatMessageListVC<ExtraData>, didTapOn attachment: _ChatMessageAttachment<ExtraData>) {
-        messageList.router.showPreview(for: attachment)
+        switch attachment.localState {
+        case .uploadingFailed:
+            channelController.client
```

Comment:
IMHO, this is behavior that we want on all message lists in app, so it should be part of message list controller not `chatVC`

Reply:
moved to `ChatMessageList` in ebc4d74abc5f051b609a29fe3eb22faf4fa8109b